modules/site-health/audit-enqueued-assets/load.php

```diff
@@ -0,0 +1,232 @@
+		set_transient( 'aea_enqueued_front_page_scripts', $enqueued_scripts, $expiration );
+	}
+}
+add_action( 'wp_print_scripts', 'perflab_aea_audit_enqueued_scripts' );
+
+/**
+ * Audit enqueued styles in the frontend. Ignore /wp-includes styles.
+ *
+ * It will save information in a transient for 12 hours.
+ *
+ * @since 1.0.0
+ */
+function perflab_aea_audit_enqueued_styles() {
+	if ( ! is_admin() && is_front_page() && current_user_can( 'view_site_health_checks' ) && false === get_transient( 'aea_enqueued_front_page_styles' ) ) {
+		global $wp_styles;
+		$enqueued_styles = array();
+		foreach ( $wp_styles->queue as $handle ) {
+			$src = $wp_styles->registered[ $handle ]->src;
+			if ( $src && ! strpos( $src, 'wp-includes' ) ) {
+				$enqueued_styles[] = array(
+					'src'  => $src,
+					'size' => perflab_aea_get_resource_file_size( perflab_aea_get_path_from_resource_url( $src ) ),
+				);
+			}
+		}
+		$expiration = apply_filters( 'perflab_aea_enqueued_front_page_styles_expiration_in_seconds', 12 * HOUR_IN_SECONDS );
+		set_transient( 'aea_enqueued_front_page_styles', $enqueued_styles, $expiration );
+	}
+}
+add_action( 'wp_print_styles', 'perflab_aea_audit_enqueued_styles' );
+
+/**
+ * Adds tests to site health.
+ *
+ * @since 1.0.0
+ *
+ * @param array $tests Site Health Tests.
+ * @return array
+ */
+function perflab_aea_add_enqueued_assets_test( $tests ) {
+	$tests['direct']['enqueued_js_assets']  = array(
+		'label' => esc_html__( 'JS assets', 'performance-lab' ),
+		'test'  => 'perflab_aea_enqueued_js_assets_test',
+	);
+	$tests['direct']['enqueued_css_assets'] = array(
+		'label' => esc_html__( 'CSS assets', 'performance-lab' ),
+		'test'  => 'perflab_aea_enqueued_css_assets_test',
+	);
+
+	return $tests;
+}
+add_filter( 'site_status_tests', 'perflab_aea_add_enqueued_assets_test' );
+
+/**
+ * Callback for enqueued_js_assets test.
+ *
+ * @since 1.0.0
+ *
+ * @return array
+ */
+function perflab_aea_enqueued_js_assets_test() {
+	/**
+	 * If the test didn't run yet, deactivate.
+	 */
+	$enqueued_scripts = perflab_aea_get_total_enqueued_scripts();
+	if ( false === $enqueued_scripts ) {
+		return array();
+	}
+
+	$result = array(
+		'label'       => esc_html__( 'Enqueued scripts', 'performance-lab' ),
+		'status'      => 'good',
+		'badge'       => array(
+			'label' => esc_html__( 'Performance', 'performance-lab' ),
+			'color' => 'blue',
+		),
+		'description' => sprintf(
+		/* translators: 1: Number of enqueued scripts. 2: "script" word. 3.Scripts size. */
+			'<p>' . esc_html__( 'The amount of %1$s enqueued %2$s (size: %3$s) is acceptable.', 'performance-lab' ) . '</p>',
+			$enqueued_scripts,
+			_n( 'script', 'scripts', $enqueued_scripts, 'performance-lab' ),
+			size_format( perflab_aea_get_total_size_bytes_enqueued_scripts() )
+		),
+		'actions'     => '',
+		'test'        => 'enqueued_js_assets',
+	);
+
+	if ( $enqueued_scripts > 30 || perflab_aea_get_total_size_bytes_enqueued_scripts() > 300000 ) {
```

Comment:
Contrary to the above filters, I think all of these thresholds may benefit from being filterable, especially since we're still unclear about them. I just want to mention that here, we can also leave it as is for now and add filters later, if we need them.

Reply:
Filters added, you are right, these make more sense.